### ISSUE TITLE
NVDA doesn’t Detect and Announce the Presence of Formulas in Lotus Symphony Spreadsheets

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#860]
**Reported by Pat on 2010-08-25 18:33**
One has to guess the presence of a formula as this feature isn’t available.

**Comment by leonardder** [comment]
Confirmed in LibreOffice. Accessibility of LibreOffice Calc isn't very good anyway.

**Comment by jcsteh** [comment]
It looks like LibreOffice Calc does now expose the presence of a formula via IAccessible2 object attributes:

> `IAccessible2 attributes: u'Formula:1+1;Note:;Shadow:;TopBorder:;BottomBorder:;LeftBorder:;RightBorder:;isdropdown:false;'`

Note the "Formula" (capital f) attribute. When there is no formula, it is still exposed, but with an empty value.

So, it should be possible to map this to NVDA's STATE_HASFORMULA in the soffice app module.

P4 for now, but good for a new developer to tackle.

**Comment by jcsteh** [comment]
Actually, let's go for p3, since it's not that difficult to implement.
